src/Migrator.sol

```diff
@@ -2,243 +2,144 @@
-
-        ERC20(address(sushiLpToken)).safeTransferFrom(msg.sender, address(this), slpBalance);
-
-        _unwrapSlp(_amountTokenMin, _amountWethMin);
-
-        _swapWethForTokenBalancer();
-
-        _depositIntoBalancerPool();
-
-        // msg.sender receives auraBPT or BPT depending on their choice to deposit into Aura pool
-        if (_stakeBpt) {
-            _depositIntoRewardsPool();
-        } else {
-            balancerPoolToken.safeTransferFrom(address(this), msg.sender, balancerPoolToken.balanceOf(address(this)));
-        }
-
-        emit Migrated(msg.sender, slpBalance, _stakeBpt);
-    }
-
-    receive() external payable {
-        require(msg.sender != address(weth), "WETH cannot be sent directly");
-    }
-
-    /*
-        Internal functions
-    */
-
+    WETH public immutable weth = WETH(payable(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2));
+    IUniswapV2Router02 public sushiRouter = IUniswapV2Router02(0xd9e1cE17f2641f24aE83637ab66a2cca9C378B9F);
+    IUniswapV2Router02 public uniswapRouter = IUniswapV2Router02(0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D);
+    
     /**
-     * @notice Get the amount WETH required to create balanced pool deposit
-     * @param _tokenAmount Amount of TOKEN to deposit
-     * @return uint256 Amount of WETH required to create 80/20 balanced deposit
+     * @inheritdoc IMigrator
      */
-    function _calculateWethWeight(uint256 _tokenAmount) internal view returns (uint256) {
-        uint256[] memory normalizedWeights = IManagedPool(address(balancerPool)).getNormalizedWeights();
-
-        uint256 amount = (((_tokenAmount * _tokenPrice()) / 1 ether) * normalizedWeights[0]) / normalizedWeights[1];
-
-        return (amount);
-    }
+    function migrate(MigrationParams calldata params) external {
+        // Validate the router being used
+        require(params.router == sushiRouter || params.router == uniswapRouter, "Invalid router");
+        
+        // If the user is staking, then the Aura pool asset must be the same as the Balancer pool token
+        if (params.stake) {
+            require(address(params.balancerPoolToken) == params.auraPool.asset(), "Invalid Aura pool");
+        }
 
-    /**
-     * @notice Unrwap SLP into TOKEN and WETH
-     */
-    function _unwrapSlp(uint256 _amountTokenMin, uint256 _amountWethMin) internal {
-        uint256 slpAmount = sushiLpToken.balanceOf(address(this));
-
-        sushiRouter.removeLiquidityETH(
-            address(token),
-            slpAmount,
-            _amountTokenMin,
-            _amountWethMin,
-            address(this),
-            block.timestamp
+        // Preemptively transfer the Uniswap pool tokens to this contract before we conduct any mutations
+        ERC20(address(params.uniswapPoolToken)).safeTransferFrom(msg.sender, address(this), params.uniswapPoolTokensIn);
+
+        IVault vault = IBalancerPoolToken(address(params.balancerPoolToken)).getVault();
```

Comment:
This needs to be an immutable field if it is a root contract.

Reply:
update to have vault be an immutable field